src/pretalx/submission/forms/submission.py

```diff
@@ -72,9 +72,14 @@ def _set_track(self, instance=None):
 
             access_code = self.access_code or getattr(instance, "access_code", None)
             if not access_code or not access_code.track:
-                self.fields["track"].queryset = self.event.tracks.filter(
+                filter = self.event.tracks.filter(
                     requires_access_code=False
                 )
+                # ensure current track selection can be preserved
+                if instance and instance.track:
```

Comment:
```suggestion
                if instance and instance.track and instance.track.requires_access_code:
```

don't really need to add that filter if the track isn't restricted, right?